src/Sulu/Bundle/AdminBundle/Resources/js/components/ImageRectangleSelection/ImageRectangleSelection.js

```diff
@@ -0,0 +1,119 @@
+export class ImageRectangleSelection extends React.PureComponent {
+    props: {
+        /** Determines the position at which the selection box is rendered at the beginning. */
+        initialSelection?: SelectionData,
+        minWidth?: number,
+        minHeight?: number,
+        onChange?: (s: SelectionData) => void,
+        src: string,
+        containerWidth: number,
+        containerHeight: number,
+    };
+
+    image: Image;
+    rounding = new RoundingNormalizer();
+    @observable imageLoaded = false;
+
+    naturalHorizontalToReal = (h: number) => h * this.imageResizedWidth / this.image.naturalWidth;
+    scaledHorizontalToNatural = (h: number) => h * this.image.naturalWidth / this.imageResizedWidth;
+    naturalVerticalToReal = (v: number) => v * this.imageResizedHeight / this.image.naturalHeight;
+    scaledVerticalToNatural = (v: number) => v * this.image.naturalHeight / this.imageResizedHeight;
+
+    naturalDataToReal(data: SelectionData): SelectionData {
+        return {
+            width: this.naturalHorizontalToReal(data.width),
+            height: this.naturalVerticalToReal(data.height),
+            left: this.naturalHorizontalToReal(data.left),
+            top: this.naturalVerticalToReal(data.top),
+        };
+    }
+
+    scaledDataToNatural(data: SelectionData): SelectionData {
+        return {
+            width: this.scaledHorizontalToNatural(data.width),
+            height: this.scaledVerticalToNatural(data.height),
+            left: this.scaledHorizontalToNatural(data.left),
+            top: this.scaledVerticalToNatural(data.top),
+        };
+    }
+
+    componentWillMount() {
+        this.image = new Image();
+        this.image.onload = action(() => this.imageLoaded = true);
+        this.image.onerror = () => log.error('Failed to preload image "' + this.props.src + '"');
+        this.image.src = this.props.src;
+    }
+
+    @computed get imageResizedHeight(): number {
+        if (this.imageTouchesHorizontalBorders()) {
+            return Math.min(this.image.naturalHeight, this.props.containerHeight);
+        } else {
+            return this.imageResizedWidth * this.image.naturalHeight / this.image.naturalWidth;
+        }
+    }
+
+    @computed get imageResizedWidth(): number {
+        if (this.imageTouchesHorizontalBorders()) {
+            return this.imageResizedHeight * this.image.naturalWidth / this.image.naturalHeight;
+        } else {
+            return Math.min(this.image.naturalWidth, this.props.containerWidth);
+        }
+    }
+
+    imageTouchesHorizontalBorders() {
+        const imageHeightToWidth = this.image.naturalHeight / this.image.naturalWidth;
+        const containerHeightToWidth = this.props.containerHeight / this.props.containerWidth;
+        return imageHeightToWidth > containerHeightToWidth;
+    }
+
+    handleRectangleSelectionChange = (data: SelectionData) => {
+        if (this.props.onChange) {
+            const onChange = this.props.onChange;
```

Comment:
Flow shows an error if I don't. I suppose Flow thinks `props.onChange` could change before calling it.